### DISCUSSION TITLE
Getting a warning when used observables in functions inside react components

### DISCUSSION THREAD

**Discussion by MetaVII** [mobxjs/mobx#3377] [Q&A] [answered]
Greetings!
This is more precisely formulated version of my previous question.
 
I have a store class, configured strictly, and observable array in it (TodoStore in the example).

I use the observable array in a function inside an observer wrapped component, and get the "[mobx] Observable being read outside a reactive context." (TodoList in example)

I would prefer to use strictly configured store. Is it any right way to do so without getting that warning?

Example below doesn't reproduce my real case, but is enough to reproduce the warning.
[example link](https://codesandbox.io/s/classic-mobx-stores-with-hooks-forked-b8ev5?file=/src/components/TodoList.js)

**Comment by urugator** [comment, marked as answer]
You're getting the warning, because `console.log` isn't synchronous, therefore it accesses `todos` outside `observer`.
Change it to `console.log(mobx.toJS(todos));` and warning is gone. The `toJS` will run as part of `observer`, subscribing to `todos` and anything within, `console.log` later prints the returned non-observable copy.

**Reply by MetaVII** [reply]
thank you for the answer
